File: TypeScript/src/accounts/paymentaccount.ts
import { AccountData } from "./accountdata";

/**
 * Refactor Exercise 2: Account
 *
 * @author Zishen Wen (F22), Deyuan Chen (S22)
 */
 abstract class PaymentAccount {
	private accountData: AccountData;

	constructor(accountName: string, accountNumber: number) {
		this.accountData = new AccountData(accountName, accountNumber);
	}

	public abstract pay(amount: number): boolean;

	public getAccountData(): AccountData {
		return this.accountData;
	}
}

export { PaymentAccount }

File: Java/src/main/java/account/PaymentAccount.java
package account;

/**
 * Refactor Exercise 2: Account
 *
 * @author Zishen Wen (F22), Deyuan Chen (S22)
 */
public abstract class PaymentAccount {

    private final AccountData accountData;

    public PaymentAccount(String accountHolder, int accountNumber) {
        this.accountData = new AccountData(accountHolder, accountNumber);
    }

    public abstract boolean pay(int amount);

    public AccountData getAccountData() {
        return this.accountData;
    }
}
